doc: Reorganization

fossison opened the pull request:
Reorganization of /doc's contents into subdirectories, grouped by use case.

Hey all,

Been using my Pinetime for a few weeks now and would like to start contributing. My first attempt at this is making a first-pass at cleaning up the /doc directory.

I've grouped the existing documentation into three categories:

- /dev: Documentation for contributing developers, including setting up a build environment for InfiniTime
- /ref: Technical reference documents for InfiniTime and the PineTime
- /user: InfiniTime user-facing documentation (How to Use InfiniTime)

Open to feedback. I'd like to continue improving documentation beyond this as well (updating user documentation, and adding a "Development with InfiniSim" page). Long-term, standardizing file names, adding indexes, etc.

About me: I've worked in the US tech industry for the past 8 years and have professional experience in Product/Project/Program Management. My programming is mostly hobby-based and spans a number of languages.

minacode commented:
Welcome to the community! 😊

Riksu9000 commented on doc/README.md:
```diff
@@ -0,0 +1,8 @@
+
+The /doc folder is organized as follows:
+
+ * [dev/](./dev): Information for developers looking to contribute
+ * [ref/](./ref): Technical references for InfiniTime and the PineTime
+ * [user/](./user): User-facing documentation (How to use InfiniTime)
```
We don't have a markdown formatter or explicit rules yet, but if we did, bullet points would be written like so.

```suggestion
- [dev/](./dev): Information for developers looking to contribute
- [ref/](./ref): Technical references for InfiniTime and the PineTime
- [user/](./user): User-facing documentation (How to use InfiniTime)
```

fossison replied:
Updated

Riksu9000 reviewed (commented):
Thanks for taking the initiative. The files have links to other files and some of them are broken by moving them. I guess the difference between ref and dev is that dev is for working inside InfiniTime, and ref is for working "around" or interfacing with InfiniTime.

fossison commented:
I've updated the PR with the following changes:

1. Updated the markdown syntax of my one list
2. Reviewed all files and fixed relative linking as needed

Note that the file `PinetimeStubWithNrf52DK.md` had a previously existing issue where it references a non-existent image.

@Riksu9000 I think that's pretty close to what I envisioned. /ref could in the future include software documentation (for example, if InfiniTime did a graphics abstraction class)

Marking this as ready for review, I should've caught all the broken links this time around.

Riksu9000 commented on README.md:
```diff
@@ -23,46 +23,46 @@ Fast open-source firmware for the [PineTime smartwatch](https://www.pine64.org/p
-- [How to implement an application](doc/code/Apps.md)
+- [InfiniTime Vision](doc/dev/InfiniTimeVision.md)
+- [Rough structure of the code](doc/dev/code/Intro.md)
+- [How to implement an application](doc/dev/code/Apps.md)
 - [Generate the fonts and symbols](src/displayapp/fonts/README.md)
-- [Tips on designing an app UI](doc/ui_guidelines.md)
+- [Tips on designing an app UI](doc/dev/ui_guidelines.md)
 - [Bootloader, OTA and DFU](bootloader/README.md)
-- [External resources](doc/ExternalResources.md)
+- [External resources](doc/ref/ExternalResources.md)
```
I wonder if this is actually the best way to organize the files. Many sections in the README includes files from both dev and ref. There are also some questions regarding the future direction of the documentation and the project in general, so it might be a bit early to be optimizing the documentation.

fossison replied:
Personally, I withheld from editing anything outside of the /doc/ folder except to fix links as I thought it'd be presumptuous to edit the readme :sweat_smile: 

What sort of questions are there? I'm more than willing to contribute to a larger-scale revamp including the readme, let me know if there's anything I could help with on that front.

JF002 commented:
Hi and welcome in the community :)

Please note that we've already started a similar initiative  [here](https://github.com/InfiniTimeOrg/InfiniTime/pull/838). There haven't been any progress recently, but the goal remains the same : Add user documentation (how to use InfiniTime), update and improve the developer and contributor documentation, and deploy it on https://docs.infinitime.io.

Right now, this PR is mostly outdated, and we should probably work on it step by step instead of trying to move the whole doc at once to the new format.
